torch/_inductor/ir.py

```diff
@@ -3792,13 +3792,24 @@ def export_extern_kernel_node(self):
         named_arguments = serializer.serialize_inputs(self.op_overload, args, kwargs)
 
         # serialize_outputs
-        if isinstance(self.outputs, (list, tuple)):
+        if isinstance(self.outputs, tuple):
+            # For tuple returns, e.g "-> (Tensor, Tensor)"
             output_arguments = [
                 export_schema.Argument.create(
                     as_tensor=export_schema.TensorArgument(name=output.get_name())
                 )
                 for output in self.outputs
             ]
+        elif isinstance(self.outputs, list):
```

Comment:
Wondering why we are differentiating ```list``` from ```tuple``` above. Particularly, why do we pass ```as_tensors``` for ```list``` but ```as_tensor``` for tuple?

Reply:
This is by design of the PT2 IR. 
TensorList is a first class data type, whereas tuple is used only when there are multiple returns. 

Consider following two cases `foo(...) -> (Tensor, Tensor)`  and `bar(...) -> Tensor[]`

In the first case, self.output is a python tuple; The output would be serialized as `[Arguemnt(asTensor="buf3"), Argument(asTensor = "buf4")]`
In the second case, self.output is a python list. The output would be serialized as `[Argument(asTensors=["buf3", "buf4"])]`. 

For more details, see description in https://github.com/pytorch/pytorch/pull/110187